Sources/OktaAnalytics/OktaAnalytics.swift

```diff
@@ -102,12 +109,199 @@ public final class OktaAnalytics: NSObject {
      - eventName: `event name` provided by client
      - withProperties: `properties/metadata` associated with event
      */
-    public static func trackEvent(_ eventName: String, withProperties: [String: String]?) {
+    public static func trackEvent(_ eventName: String, withProperties: Properties) {
         lock.readLock()
         defer { lock.unlock() }
         providers.forEach { $1.trackEvent(eventName, withProperties: withProperties) }
     }
 
+    /**
+        Starts an event scenario with the specified name.
+
+        - Parameters:
+           - eventName: The name of the event scenario to start.
+           - propertySubject: A closure that takes a `PassthroughSubject` of `Property` objects as a parameter.
+        */
+    public static func startScenario(_ scenarioName: Name, _ propertySubject: (PassthroughSubject<Property, Never>?) -> Void) -> ScenarioID {
+
+        lock.readLock()
+        defer { lock.unlock() }
+
+        var scenario = ScenarioEvent(name: scenarioName)
+        let eventScenario = EventScenario(scenario, managedContext: coreDataStack.managedContext)
+        Self.scenarios?[scenario.id] = eventScenario
+        propertySubject(eventScenario.start())
+
+        Self.providers.forEach {
+            $1.logger?.log(level: .debug, eventName: scenario.name, message: "\($0) Scenario \(scenario.name) already in flight", properties: nil, file: #file, line: #line, funcName: #function)
+        }
+        return scenario.id
+    }
+
+    /**
+        update a property scenario with the values.
+
+        - Parameters:
+           - scenarioID: unique sceario ID returned from `startScenario(_ , _)` .
+           - propertySubject: A closure that takes a `PassthroughSubject` of `Property` objects as a parameter.
+        */
+    public static func updateScenario(_ scenarioID: ScenarioID, _ propertySubject: (PassthroughSubject<Property, Never>?) -> Void) {
```

Comment:
We wanted to add a few App lifecycle utilities, like flush(). The reason  for calling this would be either (app crashed, restarted) .